### PR TITLE
[SSO/Spot] Fix spot controller failed to launch spot cluster when using SSO

### PR DESCRIPTION
**Pull request by Michaelvll**
A user reported that when using AWS SSO account, the `sky spot launch` will fail due to the controller does not have any cloud access enabled. That is a bug with our current SSO implementation, as the controller is assigned with an assumed IAM-role, which has the access to create the spot cluster and will not have static credential files, but our cloud access checking will fail when it finds there is no credential files.
Another problem with the spot controller is: the controller will keep retrying even if the all cloud disabled exception is raised, causing the spot job in STARTING status forever.

To reproduce:
1. Run in a new docker container
2. log in with SSO account
3. `sky spot launch -n test echo hi --retry-until-up`

This PR fixes:
1. The spot controller will now be able to launch the spot cluster when the user is using AWS SSO
2. Spot controller will be able to catch the all cloud disabled exception and fail early. (This is for any potential future identity problem with that exception)

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - [x] The reproducible script process.

### CODE REVIEW

File: sky/clouds/aws.py [hunk 1]
```diff
@@ -495,7 +516,9 @@ def get_current_user_identity(cls) -> Optional[List[str]]:
             # userid changed for a cluster).
             # 2. In the case where the multiple users belong to an organization,
             # those users will have different account id, so fallback works.
-            user_ids = [user_info['UserId'], user_info['Account']]
+            user_ids = [
+                user_info['UserId'], user_info['Account'], user_info['Arn']
```
Comment (concretevitamin) [inline sky/clouds/aws.py:520]: Is this change safe? Do people who upgrade past this branch need to run `status -r` explicitly?

**Review by concretevitamin — APPROVED** [review]
Thanks for the quick fix @Michaelvll! Mostly nits. Feel free to merge after.

File: sky/clouds/aws.py [hunk 2]
```diff
@@ -40,6 +41,13 @@
 DEFAULT_AMI_GB = 45
 
 
+class AWSIdentityType(enum.Enum):
+    """AWS identity type."""
+    SSO = 'sso'
+    STATIC = 'static'
+    IAM_ROLE = 'iam-role'
+
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/clouds/aws.py]: Good point! Updated. Thanks!

File: sky/clouds/aws.py [hunk 3]
```diff
@@ -433,15 +449,20 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
...
         proc = subprocess.run('aws configure list',
                               shell=True,
                               check=False,
                               stdout=subprocess.PIPE,
                               stderr=subprocess.PIPE)
         if proc.returncode != 0:
-            return False
-        return 'sso' in proc.stdout.decode().split()
+            return None
+        if AWSIdentityType.SSO.value in proc.stdout.decode():
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/clouds/aws.py:476]: Done. Thanks!

**Pull request merged**